src/redux/reducer.ts

```diff
@@ -0,0 +1,59 @@
+import {Movie} from './types';
+import {
+    CREATE_MOVIES,
+    DELETE_MOVIES,
+    GET_MOVIES,
+    UPDATE_MOVIES
+} from './constants';
+
+const initialState = { moviesList: [] as Movie[] };
+
+interface ActionType {
+    type: string;
+    payload: Movie[];
+}
+
+export interface MovieState {
+    moviesList: Movie[];
+}
+
+export const MovieReducer = (state = initialState, action: ActionType): MovieState => {
+    const deleteItem = (list, id) => {
+        if(list) {
+            return list.filter((movie) => movie.id !== id);
+        }
+        return list;
```

Comment:
Also not sure that you need this check here. But if it’s necessary you can write it like this:
```js
return list?.filter((movie) => movie.id !== id);
```